<launch>
    <!-- dataset path args -->
    <arg name="voc_path" doc="Path of DBoW dictionary"
         default="/home/yonghui/catkin_ws/src/ORB-SLAM2_DENSE/Vocabulary/ORBvoc.bin" />
    <arg name="setting_path" doc="Path of SLAM and camera config .yaml file"
         default="/home/yonghui/catkin_ws/src/ORB-SLAM2_DENSE/Examples/ROS/ORB_SLAM2_DENSE/params/TUM2.yaml" />
    <!-- orb_slam2_dense parameters args -->
    <arg name="use_odom_pub" default="true"/>
    <arg name="use_rviz" default="true" />
    <arg name="use_tf" default="true" />
    <arg name="use_frame_pub" default="true" />
    <arg name="use_pcl_pub" default="true" />
    <!-- broadcast frame id args -->
    <arg name="map_frame" default="map" />
    <arg name="odom_frame" default="odom" />
    <arg name="footprint_frame" default="camera_footprint" />
    <arg name="optical_frame" default="camera_optical" />


    <!-- orb_slam_dense -->
    <node pkg="orb_slam2_dense" type="RGBD" name="RGBD" args="$(arg voc_path) $(arg setting_path)" output="screen" >
          <!-- launch-prefix="xterm -e gdb -->
          <!-- -d /home/yonghui/catkin_ws/src/ORB-SLAM2_DENSE/ -->
          <!-- -ex 'break src/pointcloudmapping.cc:603' -->
          <!-- -ex run -->
          <!-- -/-args"> -->
        <!-- orb_slam2_dense parameters -->
        <param name="use_odom_pub" type="bool" value="$(arg use_odom_pub)" />
        <param name="use_rviz" type="bool" value="$(arg use_rviz)" />
        <param name="use_tf" type="bool" value="$(arg use_tf)" />
        <param name="use_frame_pub" type="bool" value="$(arg use_frame_pub)" />
        <param name="use_pcl_pub" type="bool" value="$(arg use_pcl_pub)" />
        <!-- frame_id parameters -->
        <param name="map_frame" type="str" value="$(arg map_frame)" />
        <param name="odom_frame" type="str" value="$(arg odom_frame)" />
        <param name="footprint_frame" type="str" value="$(arg footprint_frame)" />
        <param name="optical_frame" type="str" value="$(arg optical_frame)" />
    </node>


</launch>